DependencyInjection/Compiler/DoctrineQueryExtensionPass.php

```diff
@@ -40,7 +42,12 @@ public function process(ContainerBuilder $container)
         $extensions = call_user_func_array('array_merge', $sortedServices);
         $dataProviderDefinition = $container->getDefinition('api.doctrine.orm.data_provider');
         foreach ($extensions as $extension) {
-            $dataProviderDefinition->addMethodCall('addExtension', [$extension]);
+            $extensionClass = $container->getDefinition($extension)->getClass();
+            if (is_subclass_of($extensionClass, QueryItemExtensionInterface::class)) {
```

Comment:
Performance is not important here because this code is not executed at runtime (but only when compiling the container). Having 2 types is less ugly and as said in my previous comment, those 2 kind of extensions are very different.